internal/cmd/deploy/deploy.go

```diff
@@ -61,6 +86,33 @@ The bucket's name will be of the format rain-artifacts-<AWS account id>-<AWS reg
 		// Parse params
 		parsedParams := ListToMap("param", params)
 
+		if len(cliJSONFile) != 0 {
+			jsonFile, err := os.Open(cliJSONFile)
+			if err != nil {
+				panic(ui.Errorf(err, "error opening cli json file '%s'", cliJSONFile))
+			}
+			defer jsonFile.Close()
+
+			byteValue, err := ioutil.ReadAll(jsonFile)
+			if err != nil {
+				panic(ui.Errorf(err, "unable to read cli json file '%s'", cliJSONFile))
+			}
+
+			var cliInput cliInputJSON
+			err = json.Unmarshal([]byte(byteValue), &cliInput)
+			if err != nil {
+				panic(ui.Errorf(err, "unable to parse json in '%s'", cliJSONFile))
+			}
+
+			for k, v := range cliInput.Tags {
+				parsedTags[k] = v
+			}
+
+			for k, v := range cliInput.Parameters {
+				parsedParams[k] = v
```

Comment:
This means that tags and parameters specified in a config file will override tags and parameters supplied by `--tags` or `--params` which seems counter-intuitive.

Perhaps we should warn (or fail) on clashing tags/param keys?

Reply:
I changed the priority. The cli flags `tag` and `param` can now override the config file. When they do so, a warning will be printed. Is this what you had in mind?